pilot/pkg/kube/inject/inject.go

```diff
@@ -540,6 +540,24 @@ func flippedContains(needle, haystack string) bool {
 func InjectionData(sidecarTemplate, valuesConfig, version string, deploymentMetadata *metav1.ObjectMeta, spec *corev1.PodSpec,
 	metadata *metav1.ObjectMeta, proxyConfig *meshconfig.ProxyConfig, meshConfig *meshconfig.MeshConfig) (
 	*SidecarInjectionSpec, string, error) {
+
+	var name string
+	if metadata.Namespace == "" {
+		if deploymentMetadata.Namespace != "" {
+			name = deploymentMetadata.Namespace + "/" + metadata.Name
+		} else {
+			name = metadata.Name
+		}
+	} else {
+		name = metadata.Namespace + "/" + metadata.Name
```

Comment:
I think this line is enough, empty namesapce equals default, only happens when using istioctl to inject a object without explicitly set namespace

Reply:
Ok, updated.